test/integration/integration.h

```diff
@@ -357,6 +357,10 @@ class BaseIntegrationTest : protected Logger::Loggable<Logger::Id::testing> {
    **/
   void sendRawHttpAndWaitForResponse(int port, const char* raw_http, std::string* response,
                                      bool disconnect_after_headers_complete = false);
+  std::unique_ptr<RawConnectionDriver>
```

Comment:
newline before this line, doc comments. See below though. I'm not sure why you are using "raw HTTP" for tests you are adding for this at all. Prefer reverting all of this unless there is a good reason to add/change all of this.

Reply:
I need two functionality: 
First is to provide N filter chains in the listener. 
2nd is to verify some connection be drained and some are not drained.

The dst_ip is tackling the first requirement: 0.0.0.0:<port> listener accept 127.0.0.1:<port> and 127.0.0.2:<port>. 
RawHttp help me verify that 127.0.0.2:<port> HCM is not impacted if I remove 127.0.0.1:<port> filter chain